packages/blocks/src/api/factory.js

```diff
@@ -285,6 +286,8 @@ export function findTransform( transforms, predicate ) {
  * @return {Array} Block transforms for direction.
  */
 export function getBlockTransforms( direction, blockTypeOrName ) {
+	const { canInsertBlockType } = select( 'core/editor' );
```

Comment:
This creates a circular dependency between editor and blocks. The blocks module should not be aware of editor.

Can this be enforced from the editor module instead?

Reply:
If we could introduce a filter, something like:
```
availableTransforms = applyFilters( 'blocks.transforms', transforms );
```

then the editor module could register a filter to remove any transforms to blocks that weren't allowed. How does that sound? It seems that this filtering does need to happen here, because if the editor gets a block type that isn't allowed, it doesn't have any information about what to do with it, because the content of the paragraph block that should be created instead could live in any attribute of the block that isn't allowed.

Reply:
As an alternative to a filter, could we pass as an additional argument an array of "available" block names which can be applied as a filter for the block type? I could see this might be made more difficult by the fact there are other functions within the blocks module which call `getBlockTransforms`, it's not just called by the editor module.

My worry with a filter might be that in mind of a refactor like proposed in #7119, there may be multiple editors present which each have their own set of limited blocks, which seems difficult/impossible to manage from a single filter.

Reply:
Passing in an argument would work for cases where `editor` needs a list of valid transforms, but `pasteHandler` from `blocks` is used in the `RichText` component to handle incoming pastes, and neither of these have a dependency on `editor`, and I figure that we don't want to introduce that either. So we'd have to have a new prop on `RichText` that was the list of allowed, or denied, block types, so that the `RichText` instances for the editor instance would have the right list. Does that seem right to you? It would mean altering every instantiation of a `RichText` component, wouldn't it?